tools/ci/tests/test_pr_preview.py

```diff
@@ -252,75 +252,55 @@ def synchronize(expected_traffic, refs={}):
 
             remote_refs[ref] = revision
 
-    return child.returncode, server.actual_traffic, remote_refs
+    return (
+        method_threw,
+        github_server.actual_traffic,
+        remote_refs,
+    )
 
 
-def detect(event, expected_github_traffic, expected_preview_traffic):
-    env = {
-        'DEPLOY_TOKEN': 'c0ffee'
-    }
-    env.update(os.environ)
-    github_server = MockServer((test_host, 0), expected_github_traffic)
+def deploy(pr_num, revision, expected_github_traffic, expected_preview_traffic):
+    os.environ['GITHUB_TOKEN'] = 'c0ffee'
+
+    github_server = MockServer((TEST_HOST, 0), expected_github_traffic)
     github_port = github_server.server_address[1]
-    preview_server = MockServer((test_host, 0), expected_preview_traffic)
+    preview_server = MockServer((TEST_HOST, 0), expected_preview_traffic, reponse_body_is_json=False)
     preview_port = preview_server.server_address[1]
 
-    with temp_repo() as repo, github_server, preview_server:
-        env['GITHUB_EVENT_PATH'] = repo + '/event.json'
-
-        with open(env['GITHUB_EVENT_PATH'], 'w') as handle:
-            handle.write(json.dumps(event))
-
-        child = subprocess.Popen(
-            [
-                sys.executable,
-                subject,
-                '--host',
-                'http://{}:{}'.format(test_host, github_port),
-                '--github-project',
-                'test-org/test-repo',
-                'detect',
-                '--target',
-                'http://{}:{}'.format(test_host, preview_port),
-                '--timeout',
-                '1'
-            ],
-            cwd=repo,
-            env=env
+    method_threw = False
+    with temp_repo(), github_server, preview_server:
+        project = pr_preview.Project(
+            'http://{}:{}'.format(TEST_HOST, github_port),
+            'test-org/test-repo',
         )
-        child.communicate()
+        target = 'http://{}:{}'.format(TEST_HOST, preview_port)
+        pull_request = {'number': pr_num}
+        timeout = 1
+        try:
+            pr_preview.deploy(project, target, pull_request, revision, timeout)
+        except (pr_preview.GitHubRateLimitException, pr_preview.DeploymentFailedException):
+            method_threw = True
 
     return (
-        child.returncode,
+        method_threw,
         github_server.actual_traffic,
         preview_server.actual_traffic
     )
 
-
-def test_synchronize_zero_results():
-    expected_traffic = [
-        (Requests.get_rate, Responses.no_limit),
-        (Requests.search, (
-            200,
-            {
-                'items': [],
-                'incomplete_results': False
-            }
-        ))
-    ]
-
-    returncode, actual_traffic, remote_refs = synchronize(expected_traffic)
-
-    assert returncode == 0
-    assert same_members(expected_traffic, actual_traffic)
-
-def test_synchronize_fail_search_throttled():
+def test_update_mirror_refs_fail_search_throttled():
```

Comment:
Nit
```suggestion
def test_update_mirror_refs_fail_rate_limited():
```

This function no longer uses any API that requires "search" quota.

Reply:
Good catch! Done.